### DISCUSSION TITLE
How to calculate the SOH of the battery cell?

### DISCUSSION THREAD

**Discussion by ssingh-ipa** [pybamm-team/PyBaMM#3508] [Q&A]
I have the experimental data of a cell across its entire lifecycle, where after every 100 WLTP driving cycles, a checkup cycle (EIS measurement and CCCV charge and CC discharge ) is performed. I can divide the data into the set of beginning-of-life, mid-of-life, and end-of-life cycles, but that might not encapsulate the non-linear degradation behavior of the cell. 

Without having to give an entire cycling data and check-up cycle data as input to the pybamm code, how can I calculate the SOH of the cell? Please suggest ideas for using pybamm for SOH estimation and prediction using pybamm.

**Comment by TomTranter** [comment]
There are many things that can determine the SoH. Without simulating all the cycles you could try to fit the initial few then update the active material volume fractions and amount of cyclable lithium to capture the main effects of degradation for later cycles. It's a complex task but fitting what's changed is easier than fitting the parameters of the degradation models to determine why. @tinosulzer published a paper on using this approach

**Reply by ssingh-ipa** [reply]
Could you please provide a link to the paper?

In terms of parameters that reflect ageing and directly affect the SOH, are 'active material volume fractions' and 'amount of cyclable lithium' the main influencing factors? Is there a list of parameters that I need to fix and optimize as the number of cycles increases?

Do you suggest simulating the driving cycles or simulating the CCCV checkup cycles to capture the main effects of degradation?

**Reply by TomTranter** [reply]
This is the paper https://iopscience.iop.org/article/10.1149/1945-7111/ac3e48/meta
I would suggest the CCCV cycles as then you have the best measure of capacity
In terms of the parameters you can find example notebooks on degradation and I'd suggest having a read there.

**Reply by TomTranter** [reply]
AFAIK the algorithms developed in this paper are not all fully implemented in PyBaMM yet. Not sure @tinosulzer has had the bandwidth to do it but we are now working together at ion-works.com and you can get in touch with us there to hire us for consultancy
